homeassistant/components/qvrpro/__init__.py

```diff
@@ -0,0 +1,90 @@
+"""Support for QVR Pro NVR software by QNAP."""
+
+import logging
+
+from pyqvrpro import Client
+from pyqvrpro.client import AuthenticationError
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+from .const import CONF_EXCLUDE_CHANNELS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_EXCLUDE_CHANNELS, default=[]): vol.All(
+                    cv.ensure_list_csv, [cv.positive_int]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the QVR Pro component."""
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    host = config[DOMAIN][CONF_HOST]
+    excluded_channels = config[DOMAIN][CONF_EXCLUDE_CHANNELS]
+
+    try:
+        qvrpro = Client(user, password, host)
+    except AuthenticationError:
+        _LOGGER.error("Authentication failed")
+        return False
+
+    channel_resp = qvrpro.get_channel_list()
+
+    if "message" in channel_resp.keys():
+        if channel_resp["message"] == "Insufficient permission.":
+            _LOGGER.error("User must have Surveillance Management permission")
+            return False
```

Comment:
Please throw an error in the library when `qvrpro.get_channel_list()` is called with insufficient permissions and wrap it here with a try-catch block instead two layered if statements.